simulator/interface/gdb/mips-sim-gdb.cmake

```diff
@@ -0,0 +1,32 @@
+# Build file for MIPT-MIPS GDB interface
+# Vyacheslav Kompan
+# Copyright 2018 MIPT-MIPS
+
+
+set(GDB_SIM_COMMON_PATH ${CMAKE_CURRENT_LIST_DIR}/../../../../sim/common)
+
+# Add mipt-mips-gdb target iff we are building under GDB
+if (EXISTS ${GDB_SIM_COMMON_PATH})
+
+    set(GDB_SIM_SRCS
+            ${GDB_SIM_COMMON_PATH}/sim-utils.c
+            ${GDB_SIM_COMMON_PATH}/callback.c
+            ${GDB_SIM_COMMON_PATH}/sim-io.c)
+
+    add_library(mipt-mips-gdb STATIC ${CPPS}
```

Comment:
That's not the best way to set up dependencies, since `mipt-mips-gdb` and `mipt-mips-src` are independent entities. To speed up build, it is better to link `mipt-mips-src` into the GDB interface:

```cmake
add_library(mipt-mips-gdb STATIC ${CMAKE_CURRENT_LIST_DIR}/gdb_interface.cpp ${GDB_SIM_SRCS})
target_link_libraries(mipt-mips-gdb STATIC mipt-mips-src)
```

Reply:
It's not possible to create nested static libraries ([more 1](https://stackoverflow.com/questions/19316307/gcc-nested-libraries-and-linker-dependencies), [more 2](https://stackoverflow.com/questions/2157629/linking-static-libraries-to-other-static-libraries))

Reply:
OK, my fault — I meant not to link static libraries with static libraries, but archive one library into another:
```cmake
add_library(mipt-mips-gdb STATIC ${CMAKE_CURRENT_LIST_DIR}/gdb_interface.cpp ${GDB_SIM_SRCS})
target_link_libraries(mipt-mips-gdb mipt-mips-src)
```

Reply:
I use that syntax in every CMake project. Let me check what's wrong with it when I get to PC with Unix environment installed. Probably, we have to link `-lmipt-mips-src` to GDB as well.

> I'd prefer to leave this as is.
I prefer not to have each unit translated twice (thrice etc.), so you have to fix it.